internal/internal_task_handlers.go

```diff
@@ -1694,6 +1695,10 @@ func (i *temporalInvoker) internalHeartBeat(ctx context.Context, details *common
 		}
 	}
 
+	if err != nil {
```

Comment:
Hrmm, I wonder if it's too noisy to log canceled errors which seem like a normal thing to have happen. I guess canceled errors were logged before anyways?

Reply:
This is debug log, and I don't expect customers to enable them by default.
For us, we should record all errors. I don't worry too much about the volume here because long running activity and heartbeat is not a large scale operation.

Reply:
FWIW, I think debug log is on by default. See https://github.com/temporalio/sdk-go/blob/master/internal/log/default_logger.go where there are no conditionals on the level. I do agree we shouldn't probably have it that way, but it currently is.

Reply:
IMO it's ok to log it at the debug level. People should adjust the log level it they feel this becomes too vervose.
RecordHeartbeat is already async/best effort. So I want to make sure we record all errors so we can investigate potential issues.

Reply:
:+1: